src/data/polynomial/degree/definitions.lean

```diff
@@ -253,6 +253,20 @@ begin
   { rwa [degree_eq_nat_degree hp, with_bot.coe_lt_coe] }
 end
 
+lemma ext_iff_nat_degree_le {p q : R[X]} {n : ℕ} (hp : p.nat_degree ≤ n) (hq : q.nat_degree ≤ n) :
+  p = q ↔ (∀ i ≤ n, p.coeff i = q.coeff i)  :=
```

Comment:
Well, I am not sure I want to go down that rabbit hole. I added the result only because I needed it for #15143 and it turns out I don't need it in the end. If you want to, we can move the result to another PR and you can have a go at it if you are motivated.